### PR TITLE
Add `allow_performance` column to `osu_builds`

### PR DESCRIPTION
**Pull request by smoogipoo**
Not sure what the correct default value should be - I've gone with `false` for now but maybe it'll be `true` in the future when lazer is more "stable"?

Kinda required for https://github.com/ppy/osu-queue-score-statistics/pull/44.

### CODE REVIEW

**Comment by peppy** [comment]
I think `false` is fine. I'd want to ensure that all legacy systems creating builds don't accidentally set it to `true`.

File: database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('osu_builds', function (Blueprint $table) {
+            $table->boolean('allow_performance')->default(false);
```
Comment (peppy) [inline database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php:20]:
```suggestion
            $table->boolean('allow_performance')->default(false)->after('allow_bancho');
```

Just for my own sanity.

Also as discussed, let's add an index on this column.

**Review by peppy — CHANGES_REQUESTED** [review]

File: database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php [hunk 2]
```diff
@@ -30,6 +31,7 @@ public function down()
     {
         Schema::table('osu_builds', function (Blueprint $table) {
             $table->dropColumn('allow_performance');
+            $table->dropIndex('allow_performance');
```
Comment (peppy) [inline database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php:34]: Not sure if order matters but this feels backwards 🤔 
I guess laravel is smart enough to run correctly?

Comment (smoogipoo) [reply database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php:34]: :| I looked at other migrations in reference to this, but yes this does indeed fail. As should e.g. https://github.com/ppy/osu-web/blob/2ffc17cc6268898f1f9e19aa304436fcbaa9490c/database/migrations/2021_08_30_153214_add_track_id_to_beatmapsets.php#L30-L36

Comment (peppy) [reply database/migrations/2022_07_08_085310_add_allow_performance_to_osu_builds.php]: Good thing we don't rollback migrations.. ever 🙈

**Review by nanaya — APPROVED** [review]

**Pull request merged**